llvm/include/llvm/Support/DebugLog.h

```diff
@@ -76,6 +76,13 @@ namespace llvm {
 #define __LLVM_FILE_NAME__ ::llvm::impl::getShortFileName(__FILE__)
 #endif
 
+#define LDBG_OS(name, LEVEL)                                                   \
```

Comment:
The name is a bit confusing as it could also have been seen as debug type one. Could it be done so that it is inline? (and given inheritance, they could just assign to variable). Then the below could become

printAsTextualPipeline(LDBG_OS(1), /*pretty=*/false));

and 

raw_ostream& os = LDBG_OS(type, level); ?

Reply:
How is it now?